src/Translation/Extractor/JsTranslationExtractor.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Translation\Extractor;
+
+use Symfony\Component\Translation\Extractor\ExtractorInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+
+class JsTranslationExtractor implements ExtractorInterface
+{
+    public function __construct(string $directory)
+    {
+        $this->directory = $directory;
+    }
+
+    private string $prefix = '';
+
+    public function setPrefix(string $prefix): void
+    {
+        $this->prefix = $prefix ?: '';
+    }
+
+    public function extract($directory, MessageCatalogue $catalogue): void
+    {
+        $directory = $this->directory;
```

Comment:
SonarCloud [complains](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=1716&id=NatLibFi_Skosmos&open=AZO2xxKtw_CSfO7Da1EM) about overriding the $directory parameter.

I don't think there's any need to assign `$directory = $this->directory;`, the code below could just directly access `$this->directory`, avoiding the problem.

Reply:
_$directory = $this->directory;_ removed and it works.